static/js/ducks/objectTags.js

```diff
@@ -0,0 +1,60 @@
+import messageHandler from "baselayer/MessageHandler";
+import * as API from "../API";
+import { showNotification } from "baselayer/components/Notifications";
+import store from "../store";
+
+const FETCH_TAG_OPTIONS = "skyportal/FETCH_TAG_OPTIONS";
+const FETCH_TAG_OPTIONS_OK = "skyportal/FETCH_TAG_OPTIONS_OK";
+
+const CREATE_TAG_OPTION = "skyportal/CREATE_TAG_OPTION";
+const CREATE_TAG_OPTION_OK = "skyportal/CREATE_TAG_OPTION_OK";
+
+const FETCH_OBJECT_TAGS = "skyportal/FETCH_OBJECT_TAGS";
+
+const ADD_OBJECT_TAG = "skyportal/ADD_OBJECT_TAG";
+
+const DELETE_OBJECT_TAG = "skyportal/DELETE_OBJECT_TAG";
+
+export function fetchTagOptions() {
+  return API.GET("/api/objtagoption", FETCH_TAG_OPTIONS);
+}
+
+export function createTagOption(data) {
+  return API.POST("/api/objtagoption", CREATE_TAG_OPTION, data);
+}
+
+export function fetchObjectTags(objectId) {
+  return API.GET(`/api/objtag?obj_id=${objectId}`, FETCH_OBJECT_TAGS);
+}
+
+export function addObjectTag(data) {
+  return API.POST("/api/objtag", ADD_OBJECT_TAG, data);
+}
+
+export function deleteObjectTag(data) {
+  return API.DELETE(`/api/objtag/${data.id}`, DELETE_OBJECT_TAG, data);
+}
+
+messageHandler.add((actionType, payload, dispatch) => {
```

Comment:
but it doesn't look like you are even using the output of this right? If you want to update the object tags in place, might want to listen for an `FETCH_OBJECT_TAGS_OK` event in the `source.js` ducks to update a source's tags?

Reply:
@camilledzt you still don't need/use this websocket thing right? Like never does your backend send a `skyportal/FETCH_OBJECT_TAGS` action am I correct? It only sends a `REFRESH_SOURCE` action I believe? If so, please remove this before we can merge